run_wml_tests

```diff
@@ -1,391 +1,214 @@
+    PASS = 0
+    FAIL = 1
+    TIMEOUT = 2
+    FAIL_LOADING_REPLAY = 3
+    FAIL_PLAYING_REPLAY = 4
+
+class TestCase:
+    """Represents a single line of the wml_test_schedule."""
+    def __init__(self, status, name):
+        self.status = status
+        self.name = name
+
+    def __str__(self):
+        return "TestCase<{status}, {name}>".format(status=self.status, name=self.name)
+
+class TestListParser:
+    """Each line in the list of tests should be formatted:
+        <expected return code><space><name of unit test scenario>
+
+    For example:
+        0 test_functionality
+
+    Lines beginning # are treated as comments.
+    """
+    def __init__(self, options):
+        self.verbose = options.verbose
+        self.filename = options.list
+
+    def get(self):
+        status_name_re = re.compile(r"^(\d+) ([\w-]+)$")
+        test_list = []
+        for line in open(self.filename, mode="rt"):
+            line = line.strip()
+            if line == "" or line.startswith("#"):
+                continue
+
+            x = status_name_re.match(line)
+            if x is None:
+                print("Could not parse test list file: ", line)
+
+            t = TestCase(UnitTestResult(int(x.groups()[0])), x.groups()[1])
+            if self.verbose > 1:
+                print(t)
+            test_list.append(t)
+        return test_list
+
+class WesnothRunner:
+    def __init__(self, options):
+        self.verbose = options.verbose
+        if options.path is None:
+            path = os.path.split(os.path.realpath(sys.argv[0]))[0]
+        elif options.path in ["XCode", "xcode", "Xcode"]:
+            import glob
+            path_list = []
+            for build in ["Debug", "Release"]:
+                pattern = os.path.join("~/Library/Developer/XCode/DerivedData/Wesnoth*",
+                    build, "Build/Products/Release/Wesnoth.app/Contents/MacOS/")
+                path_list.extend(glob.glob(os.path.expanduser(pattern)))
+            if len(path_list) == 0:
+                raise FileNotFoundError("Couldn't find your xcode build dir")
+            if len(path_list) > 1:
+                # seems better than choosing one at random
+                raise RuntimeError("Found more than one xcode build dir")
+            path = path_list[0]
+        else:
+            path = options.path
+        path += "/wesnoth"
+        if options.debug_bin:
+            path += "-debug"
+        self.common_args = [path]
+        if options.strict_mode:
+            self.common_args.append("--log-strict=warning")
+        if options.clean:
+            self.common_args.append("--noaddons")
+        if options.additional_arg is not None:
+            self.common_args.extend(options.additional_arg)
+        self.timeout = options.timeout
+        if self.verbose > 1:
+            print("Options that will be used for all Wesnoth instances:", repr(self.common_args))
+
+    def _run_with_rerun_for_sdl_video(args, timeout):
```

Comment:
This function appears to be unused.

Reply:
Thanks for catching this, now fixed (although the function has moved up to file-scope).